src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -942,6 +935,22 @@ private void ScanSyntaxToken(ref TokenInfo info)
             }
         }
 
+        private bool TryScanInterpolatedString(ref TokenInfo info)
+        {
+            if (TextWindow.PeekChar(1) == '"')
+            {
+                this.ScanInterpolatedStringLiteral(isVerbatim: false, ref info);
+                return true;
+            }
+            else if (TextWindow.PeekChar(1) == '@' && TextWindow.PeekChar(2) == '"')
+            {
+                this.ScanInterpolatedStringLiteral(isVerbatim: true, ref info);
+                return true;
+            }
+
+            return false;
+        }
```

Comment:
by extracting and sharing the logic here life gets much easier for raw-string literals as there's only one source of truth on determining if we should parse/sub-parse an interpolated string.